Jenkinsfile

```
@@ -147,6 +147,23 @@ def fault_test_tag() {
     return ""
 }
 
+def release_candidate() {
+    if env.ON_TAG {
+        if env.RC_IN_TAG {
```

These really don't need to be environment variables.  They can just be regular groovy variables, or more likely functions -- or just determined in this function -- since they need to be executed in the context of a code checkout which only happens inside of stages.

These also should be able to be done in a single step since the `TAGS` file contains the tag to be applied.  And indeed, since the tag isn't yet applied in the case of a PR, you should look only at the `TAGS` file and not the tag in git.

```suggestion
    if sh(label: "Determine if building (a PR of) an RC",
          script: "git diff-index --name-only HEAD^ | grep -q TAG" && " +
                  "grep -i '[0-9]rc[0-9] TAG',
          returnStatus: true) {
```